ui/app/templates/jobs/index.hbs

```diff
@@ -12,8 +12,8 @@
     </div>
     {{#if (media "isMobile")}}
       <div class="toolbar-item is-right-aligned">
-        {{#if (can "run job")}}
-          <LinkTo @route="jobs.run" data-test-run-job class="button is-primary">Run Job</LinkTo>
+        {{#if (can "run job" namespace=this.qpNamespace)}}
+          <LinkTo @route="jobs.run" @query={{hash namespace=this.qpNamespace}} data-test-run-job class="button is-primary">Run Job</LinkTo>
```

Comment:
I can't comment on the ability file, but I think we need to add an extra check to allow running jobs if the wildcard namespace (`*`) is selected.

For example, a policy like this:
```hcl
namespace "dev" {
  policy = "write"
}
```

Should allow the `Run Job` button to be enabled. If the input job has a namespace that is not `dev`, then the job submission will fail. But the user should have the ability to at least open the job submit page if the `*` namespace is selected and their token has a `submit-job` capability in _any_ namespace.

Reply:
Buck disagreed with that, I think. Which is why we had this issue:  https://github.com/hashicorp/nomad/issues/10545.

Buck's opinions were that routes and views shouldn't be accessible if you don't have permissions. You shouldn't think you can go somewhere that you actually can't and then fail.

Reply:
But in the example that I gave you **do** have permission to run jobs in the `dev` namespace. If you are looking at `All (*)` namespaces, this includes `dev`, so you should be able to click on `Run Job`. Currently the button is disabled.